onnxruntime/contrib_ops/cpu/murmur_hash3.cc

```diff
@@ -171,34 +171,42 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
   Tensor* output_tensor = ctx->Output(0, input_shape);
 
   const MLDataType keys_type = keys->DataType();
-  const int input_element_bytes = static_cast<int>(keys->DataType()->Size());
-  const int output_element_bytes = static_cast<int>(output_tensor->DataType()->Size());
+  const bool is_string = keys_type == DataTypeImpl::GetType<std::string>();
+  if (!is_string && (DataTypeImpl::GetType<int32_t>() != keys_type) && 
+      (DataTypeImpl::GetType<uint32_t>() != keys_type)) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED, "Type not supported.");
+  }
+
+  const auto input_element_bytes = keys->DataType()->Size();
+  const auto output_element_bytes = output_tensor->DataType()->Size();
   const int64_t input_count = input_shape.Size();
-  for (int i = 0; i < input_count; ++i) {
-    if (DataTypeImpl::GetType<std::string>() == keys_type) {
-      auto input = keys->DataRaw();
-      auto output = output_tensor->MutableDataRaw();
-      auto input_string = reinterpret_cast<const std::string*>(input)[i];
-      MurmurHash3_x86_32(input_string.c_str(),
-                         static_cast<int>(input_string.length()),
+  // Output type is inferred by the inference function
+  ORT_ENFORCE(sizeof(uint32_t) == output_element_bytes, "Invalid assumption of output element size");
+  auto output = reinterpret_cast<uint32_t*>(output_tensor->MutableDataRaw());
```

Comment:
Do you need the *Raw version? output_tensor->MutableData<uint32_t>() won't work? The non-raw version does type checking too. Also, the coding guidelines mention "Qualify usages of 'auto' with 'const', '*', '&' and '&&' where applicable to more clearly express the intent".

Reply:
It probably would work, but may be misleading due to the fact that input can also be init32_t.

---
In reply to: [338842252](https://github.com/microsoft/onnxruntime/pull/2257#discussion_r338842252) [](ancestors = 338842252)

Reply:
Ok, regarding the second point: would've been good if the coding guidelines were followed.